packages/icons/src/ux-icon-map.ts

```diff
@@ -1,24 +1,47 @@
-export interface UxIconReg {
+import { getLogger } from 'aurelia-logging';
+
+const logger = getLogger('ux-icon-map');
+
+export interface UxIconRegObject {
   name: string;
-  material: string;
+  svg: string;
 }
 
+export type UxIconRegArray = [string, string, number?, number?];
+
 export class UxIconMap {
 
   private map: {
     [key: string]: string;
   } = {};
 
-  public registerIcon(name: string, material: string) {
-    this.map[name] = material;
+  public registerIcon(nameOrIcon: string | UxIconRegArray | UxIconRegObject, svg?: string) {
+    let name: string;
+    if (Array.isArray(nameOrIcon) && nameOrIcon.length >= 2) {
+      svg = this.buildSvg(nameOrIcon);
+      name = nameOrIcon[0];
+    } else if (typeof nameOrIcon === 'object' && !Array.isArray(nameOrIcon)) {
+      svg = nameOrIcon.svg;
+      name = nameOrIcon.name;
+    } else if (typeof nameOrIcon === 'string' && typeof svg === 'string') {
+      name = nameOrIcon;
+    } else {
+      logger.warn('Invalid icon', nameOrIcon);
+      return;
+    }
+    this.map[name] = svg;
   }
 
-  public registerIcons(icons: Array<UxIconReg>) {
-    if (Array.isArray(icons) && icons.length > 0 && typeof icons[0].name === 'string' && typeof icons[0].material === 'string') {
-      icons.map(icon => this.registerIcon(icon.name, icon.material));
+  public registerIcons(icons: Array<UxIconRegArray | UxIconRegObject>) {
+    if (Array.isArray(icons)) {
```

Comment:
Is this check needed?

Reply:
I placed this check because this function is called from the plugin config. When a user configure the plugin and pass an IconMap to the plugin it doesn’t have type checking (as far as I know). This test is a small guard to avoid a blocking error if an incorrect set is passed. 

However I think that if I test I should console a warning in the ‘else’ block.

Otherwise I could also remove the test... other ideas?

Reply:
I think we can leave it as is. That's fine.
Ok, do you feel it's ready to merge @ben-girardet ?

Reply:
@EisenbergEffect  Just want to extract the icon default in the class level property and then we can merge. We should also prepare a doc or info for the breaking change as the icon set will not be imported by default. What would be the best way to do this?

Reply:
If you want to add a section to the PR, I can include that in release notes.